src/OrchardCore/OrchardCore.DisplayManagement/RazorPages/Page.cs

```diff
@@ -226,6 +226,98 @@ public TagBuilder Tag(dynamic shape, string tag)
             return Shape.GetTagBuilder(shape, tag);
         }
 
+        /// <summary>
+        /// Check if a zone is defined in the layout or it has items.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        public bool IsSectionDefined(string name)
+        {
+            // We can replace the base implementation as it can't be called on a view that is not an actual MVC Layout.
+
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            var zone = ThemeLayout[name];
+
+            return zone != null && zone.Items.Count > 0;
+        }
+
+        /// <summary>
+        /// Renders a zone from the layout.
+        /// </summary>
+        /// <param name="name">The name of the zone to render.</param>
+        public IHtmlContent RenderSection(string name)
+        {
+            // We can replace the base implementation as it can't be called on a view that is not an actual MVC Layout.
+
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            return RenderSection(name, required: true);
+        }
+
+        /// <summary>
+        /// Renders a zone from the layout.
+        /// </summary>
+        /// <param name="name">The name of the zone to render.</param>
+        /// <param name="required">Whether the zone is required or not.</param>
+        public IHtmlContent RenderSection(string name, bool required)
```

Comment:
No. just don't create this method and let the users take the responsibility to call GetAwaiter(). We should not encourage that, even more at this level that can be highly concurrent.

Reply:
I can remove this method from Page.cs but then the behavior will be inconstant between MVC views and RazorPages. 

As we can't remove this method from RazorPage.cs because it's required there to hide base method.